src/main/java/dev/jbang/dependencies/ArtifactResolver.java

```diff
@@ -155,11 +155,13 @@ private ArtifactResolver(Builder builder) {
 		settings = newEffectiveSettings();
 		session = newSession(system, settings);
 		configureSession(session, settings);
+		List<RemoteRepository> partialRepos;
 		if (builder.repositories != null) {
-			repositories = builder.repositories.stream().map(this::toRemoteRepo).collect(Collectors.toList());
+			partialRepos = builder.repositories.stream().map(this::toRemoteRepo).collect(Collectors.toList());
 		} else {
-			repositories = newRepositories();
+			partialRepos = newRepositories();
 		}
+		repositories = system.newResolutionRepositories(session, partialRepos);
```

Comment:
Great find! Where did you encounter this solution?

Reply:
Found by doing find references until I found where repos auth gets set. 

So...pure luck? :)

Reply:
Well obviously a bit more than luck. A bit like stumbling around in the dark but at least you brought one of those tiny keychain flashlights 😄